learning_assistant/views.py

```diff
@@ -149,3 +154,75 @@ def get(self, request, course_run_id):
         }
 
         return Response(status=http_status.HTTP_200_OK, data=data)
+
+
+class LearningAssistantMessageHistoryView(APIView):
+    """
+    View to retrieve the message history for user in a course.
+
+    This endpoint returns the message history stored in the LearningAssistantMessage table in a course
+    represented by the course_key, which is provided in the URL.
+
+    Accepts: [GET]
+
+    Path: /learning_assistant/v1/course_id/{course_key}/history/{message_count}
+
+    Parameters:
+        * course_key: the ID of the course
+
+    Responses:
+        * 200: OK
+        * 400: Malformed Request - Course ID is not a valid course ID.
+    """
+
+    authentication_classes = (SessionAuthentication, JwtAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, course_run_id, message_count):
+        """
+        Given a course run ID, retrieve the message history for the corresponding user.
+
+        The response will be in the following format.
+
+            {
+                'status': <str>,
+                'data: {
+                    'detail': <str>,
+                    OR
+                    'message_history': <str object>,
+                },
+            }
+        """
+        try:
+            courserun_key = CourseKey.from_string(course_run_id)
+        except InvalidKeyError:
+            return Response(
+                status=http_status.HTTP_400_BAD_REQUEST,
+                data={'detail': 'Course ID is not a valid course ID.'}
+            )
+
+        if not learning_assistant_enabled(courserun_key):
+            return Response(
+                status=http_status.HTTP_403_FORBIDDEN,
+                data={'detail': 'Learning assistant not enabled for course.'}
+            )
+
+        # If user does not have an enrollment record, or is not staff, they should not have access
+        # NOTE: Do we need this? Do we currently not have any plans to show message history to audit mode learners?
+        user_role = get_user_role(request.user, courserun_key)
+        enrollment_object = CourseEnrollment.get_enrollment(request.user, courserun_key)
+        enrollment_mode = enrollment_object.mode if enrollment_object else None
+        if (
+            (enrollment_mode not in CourseMode.VERIFIED_MODES)
+            and not user_role_is_staff(user_role)
+        ):
+            return Response(
+                status=http_status.HTTP_403_FORBIDDEN,
+                data={'detail': 'Must be staff or have valid enrollment.'}
+            )
+
+        course_id = get_course_id(course_run_id)
+        user = request.user
+        data = get_message_history(course_id, user, message_count)
```

Comment:
Before being returned, this should be serialized (like https://github.com/edx/edx-exams/blob/d8cf60a98bd86be4cab31f16ae5969a7c52ef396/edx_exams/apps/api/v1/views.py#L227). This gives us better control over what we return.